dev/source/docs/ardupilot-on-linux-porting.rst

```diff
@@ -0,0 +1,142 @@
+.. _ardupilot-on-linux-porting:
+
+=========================================
+Porting to a new flight controller board
+=========================================
+
+This section is for topics related to running ArduPilot directly on Linux boards.
+See this :ref:`page for porting to boards running ChibiOS <porting>`.
+
+.. note:: Any firmware customization of ArduPilot code must abide by the terms of the GPL3.0+ open source code license. ArduPilot also reminds developers and manufacturers to adhere to the appropriate trademark and copyright laws when developing new autopilots
+
+.. image:: ../../../images/gpl3.png
+    :target: https://www.gnu.org/licenses/gpl-3.0.en.html
+
+Consider joining the `ArduPilot Discord Chat <https://ardupilot.org/discord>`__, to speak with other developers about this topic under the HAL-DEV, #Linux channel.
+
+Step 1 - getting started
+========================
+
+- Determine which microprocessor the new flight controllers uses. If it is used in a board we already support, for example BCM283x used in Raspberry Pi or TI Sitara/OMAP used in BeagleBones, then the port should be relatively straight forward.
+- Choose a new board name.
+- Reserve a new HAL_BOARD_SUBTYPE_LINUX_$(NEW_BOARD_NAME) in libraries/AP_HAL/AP_HAL_Boards.h
```

Comment:
probably need to add that ardupilot linux required spidev and i2cdev to compile for now